### ISSUE TITLE
Remove twisted.words.protocols.toc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4363]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| amacleod reported|
|-|-|
|Trac ID|trac#4363|
|Type|task|
|Created|2010-03-11 01:17:55Z|
|Branch|https://github.com/twisted/twisted/tree/remove-toc-4363|

We should get rid of twisted.words.protocols.toc.  It has been deprecated in [#3580](https://github.com/twisted/twisted/issues/3580).  Per exarkun in [[#4007](https://github.com/twisted/twisted/issues/4007)](https://github.com/twisted/twisted/issues/4007), the earliest that toc can be removed would be after December of 2010.

As soon as this ticket is resolved, also do #4007.

<details><summary>Searchable metadata</summary>

```
trac-id__4363 4363
type__task task
reporter__amacleod amacleod
priority__low low
milestone__ 
branch__branches_remove_toc_4363 branches/remove-toc-4363
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1268270275000000 1268270275000000
changetime__1295606708000000 1295606708000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Closed [#2236](https://github.com/twisted/twisted/issues/2236) as a result of this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [30521]) Branching to 'remove-toc-4363'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [30522]) Remove deprecated t.w.p.toc. refs [#4363](https://github.com/twisted/twisted/issues/4363)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

[#4007](https://github.com/twisted/twisted/issues/4007) is also up for review.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli set owner to @thijstriemstra|
|-|-|

1.You should also remove ```twisted.words.toctap``` (since that is a plugin for setting up a TOC server) and ```twisted.words.im.tocsupport```.
 1. Once you remove ```twisted.words.toctap```, could you also remove the ```twisted.plugins.twisted_words``` reference to `toctap` and update the copyright?

Thank you for working on this thijs!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

(In [30539]) Remove toctap, refs [#4363](https://github.com/twisted/twisted/issues/4363)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Doc builder looks [ok](http://buildbot.twistedmatrix.com/builders/documentation/builds/884).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [30540]) update copyright, refs [#4363](https://github.com/twisted/twisted/issues/4363)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli set owner to @thijstriemstra|
|-|-|

It is 2011 now, but other than that, it looks good and the [tests pass](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-toc-4363).  Please merge after the change from 2010 to 2011.  :)  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [30543]) Merge remove-toc-4363: Remove twisted.words.protocols.toc, that was largely non-working and useless since AOL disabled TOC on their AIM network.

Author: thijs
Reviewer: cyli
Fixes: [#4363](https://github.com/twisted/twisted/issues/4363)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
